ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java

```diff
@@ -235,11 +235,8 @@ public Optional<TransactionSimulatorResult> processWithWorldUpdater(
             ? callParams.getGasLimit()
             : blockHeaderToProcess.getGasLimit();
     if (rpcGasCap > 0) {
-      final long gasCap = rpcGasCap;
-      if (gasCap < gasLimit) {
-        gasLimit = gasCap;
-        LOG.info("Capping gasLimit to " + gasCap);
-      }
+      gasLimit = rpcGasCap;
+      LOG.info("Capping gasLimit to " + rpcGasCap);
```

Comment:
so no longer comparing gasCap to gasLimit, just if > zero?

Reply:
yes, it is essentially an override value from the cli.  @matkt can speak to the specifics about why we would raise the cap beyond the requested limit, but I think it has something to do with simulating transactions that will exceed the anticipated gas, but not exceed some predefined hard cap (DoS vector).

Reply:
yes it is need in order to allow user setting a high gasLimit for ethcall